### PR TITLE
Arrow endpoint highlighting (now on dev)

### LINKED ISSUE #12: Indicate items at both ends of arrows when they are selected
Currently, when arrows are selected, the line of the selected arrow will become wider, which makes it easier to follow the line.
It would be even easier to see what items are at the end of each line if the items on each end would have a similar indication (maybe even colored)

### PR DESCRIPTION
**Pull request by nonbinarysolo**
This resolves #12 - it's the same changes as in #41, but now on the correct branch. Arrows will highlight connected objects when selected and restore the objects' pens when deselected. Also updated the coding style to fit the project specifications.

### CODE REVIEW

File: graphics-items/arrow.cpp [hunk 1]
```diff
@@ -71,12 +71,15 @@ const QPen Arrow::RedArrowheadPen(QColor(255, 0, 0), 2, Qt::SolidLine, Qt::Round
 
 Arrow::Arrow(
   QGraphicsPolygonItem* startItem, QGraphicsPolygonItem* endItem,
-   const QPen& highlightBodyPen, const QPen& highlightArrowBasePen,
-   const QPen& highlightArrowTipPen, AeraVisualizerScene* parent)
-: QGraphicsLineItem(),
+  const QPen& highlightBodyPen, const QPen& highlightArrowBasePen,
```
Comment (jefft0) [inline graphics-items/arrow.cpp:74]: Restore whitespace.

File: graphics-items/arrow.cpp [hunk 2]
```diff
@@ -71,12 +71,15 @@ const QPen Arrow::RedArrowheadPen(QColor(255, 0, 0), 2, Qt::SolidLine, Qt::Round
...
+  const QPen& highlightArrowTipPen, AeraVisualizerScene* parent)
```
Comment (jefft0) [inline graphics-items/arrow.cpp:75]: Restore whitespace.

File: graphics-items/arrow.cpp [hunk 3]
```diff
@@ -71,12 +71,15 @@ const QPen Arrow::RedArrowheadPen(QColor(255, 0, 0), 2, Qt::SolidLine, Qt::Round
...
+  : QGraphicsLineItem(),
```
Comment (jefft0) [inline graphics-items/arrow.cpp:76]: Restore whitespace.

File: graphics-items/arrow.cpp [hunk 4]
```diff
@@ -208,7 +246,7 @@ QPointF Arrow::intersectItem(const QLineF& line, const QGraphicsPolygonItem& ite
 {
   QPolygonF polygon(item.polygon());
   QPointF itemPos(item.pos());
-
+
```
Comment (jefft0) [inline graphics-items/arrow.cpp:249]: Restore whitespace.

File: graphics-items/arrow.hpp [hunk 5]
```diff
@@ -167,6 +167,9 @@ class Arrow : public QGraphicsLineItem
   QPolygonF arrowTip_;
   QPen arrowBasePen_;
   QPen arrowTipPen_;
+  bool wasSelected_;        // Use this to track a "falling edge" on the select state
+  QPen startItemLastPen_;   // Used to restore pen settings after deselection
+  QPen endItemLastPen_;     //
```
Comment (jefft0) [inline graphics-items/arrow.hpp:172]: Comment has no text. Either remove the comment `//` or add text.

**Comment by jefft0** [comment]
That was some fast work! It looks good to me. We can do a demo at the meeting today. I made some very minor code comments about unintended changes to whitespace. We can discuss at the meeting if it is work correcting them (with a rebase and force push).

**Comment by nonbinarysolo** [comment]
Okay, sounds good!

**Pull request merged**